manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -4,21 +4,223 @@
+}
+
+// StepSink holds information to where the target data will be written:
+// it could be assetID of an asset specified in fybrikapplication or of an asset created
+// by fybrik control-plane
+type StepSink struct {
+	// AssetID identifies the target asset of this step
+	// +required
+	AssetID string `json:"assetId"`
+}
+
+// StepParameters holds the parameters to the module
+// that is deployed in this step
+type StepParameters struct {
+	// +optional
+	Source *StepSource `json:"source,omitempty"`
+
+	// +optional
+	Sink *StepSink `json:"sink,omitempty"`
+
+	// +optional
+	API *Service `json:"api,omitempty"`
+
+	// Actions are the data transformations that the module supports
+	// +optional
+	Actions []serde.Arbitrary `json:"action,omitempty"`
+}
+
+// DataFlowStep contains details on a single data flow step
+type DataFlowStep struct {
+	// Name of the step
+	// +required
+	Name string `json:"name"`
+
+	// Name of the cluster this step is executed on
+	// +required
+	Cluster string `json:"cluster"`
+
+	// Template is the name of the template to execute the step
+	// The full details of the template can be extracted from Plotter.spec.templates
+	// list field.
+	// +required
+	Template string `json:"template"`
+
+	// Step parameters
+	// TODO why not flatten the parameters into this data flow step
+	// +optional
+	Parameters *StepParameters `json:"parameters,omitempty"`
+}
+
+// SubFlowTrigger indicates the trigger for this subflow
+// +kubebuilder:validation:Enum=workload;init;timer
+type SubFlowTrigger string
+
+const (
+	// Copy flow trigger
+	CopyTrigger SubFlowTrigger = "init"
+
+	// Read flow trigger
+	ReadTrigger SubFlowTrigger = "workload"
+
+	// Write flow trigger
+	WriteTrigger SubFlowTrigger = "timer"
+)
+
+// Subflows is a list of data flows which are originated from the same data asset
+// but are triggered differently (e.g., one upon init
+// trigger and one upon workload trigger)
+type SubFlow struct {
+	// Name of the SubFlow
+	// +required
+	Name string `json:"name"`
+
+	// Type of the flow (e.g. read)
+	// +required
+	FlowType DataFlow `json:"flowType"`
+
+	// Triggers
+	// +required
+	Triggers []SubFlowTrigger `json:"triggers"`
+
+	// Steps defines a series of sequential/parallel data flow steps
+	// +required
+	Steps [][]DataFlowStep `json:"steps" protobuf:"bytes,11,opt,name=steps"`
+}
+
+// Flows is the list of data flows driven from fybrikapplication:
+// Each element in the list holds the flow of the data requested in fybrikapplication.
+type Flow struct {
+	// Name of the flow
+	// +required
+	Name string `json:"name"`
+
+	// Type of the flow (e.g. read)
+	// +required
+	FlowType DataFlow `json:"flowType"`
+
+	// AssetID indicates the data set being used in this data flow
+	// +required
+	AssetID string `json:"assetId"`
+
+	// +required
+	SubFlows []SubFlow `json:"subFlows"`
+}
+
+// ModuleInfo is a copy of M4DModule Custom Resource.  It contains information
+// to instantiate resource of type M4DModule.
```

Comment:
Can we reuse the existing FybrikModuleSpec instead of duplicating the structure?

Reply:
@simanadler We could. what should we do with `kind` and `name` fields in `ModuleInfo`? does it make sense to use FybrikModuleSpec.description as name and drop the kind field, or to have `FybrikModuleSpec` as a field in `ModuleInfo` ?

Reply:
I don't think we are using 'kind' anywhere in the code.
Regarding ModuleInfo, we don't need all the information from the ModuleSpec. The fields that we need are 
- Name
- Type
- Chart
- Scope
- Capability (read, copy, write)
- API (?)

Reply:
@shlomitk1 as discussed, the ModuleInfo is updated.

Reply:
Following a talk with @shlomitk,1 ModuleInfo was updated to hold the following information which is specific to the capability that the module was chosen for: 

* Name
* Type
* Chart
* Scope
* API